# basic_nav/config/costmap_params.yaml

###############################################################################
# Global costmap
###############################################################################
global_costmap:
  global_frame: map
  robot_base_frame: base_link
  update_frequency: 5.0
  publish_frequency: 2.0
  static_map: false
  transform_tolerance: 2.0
  
  # Размер карты затрат
  width: 100.0
  height: 100.0
  
  # Смещение начала карты затрат (важно: начало должно включать точку (0,0))
  origin_x: -50.0
  origin_y: -50.0
  resolution: 0.05
  
  # Параметры инфляции
  inflation_radius: 2.0
  cost_scaling_factor: 5.0
  
  plugins:
    - {name: static_layer, type: "costmap_2d::StaticLayer"}
    - {name: obstacle_layer, type: "costmap_2d::ObstacleLayer"}
    - {name: inflation_layer, type: "costmap_2d::InflationLayer"}

static_layer:
  map_topic: map
  subscribe_to_updates: true
  track_unknown_space: true
  use_maximum: false
  trinary_costmap: false
  lethal_cost_threshold: 80

obstacle_layer:
  observation_sources: scan
  scan:
    data_type: LaserScan
    topic: /scan
    sensor_frame: lidar_link
    marking: true
    clearing: true
    min_obstacle_height: 0.0
    max_obstacle_height: 2.0
    expected_update_rate: 5.0
    observation_persistence: 0.0
  obstacle_range: 15.0
  raytrace_range: 18.0
  track_unknown_space: true
  combination_method: 1  # 1 = maximum

inflation_layer:
  inflation_radius: 1.5
  cost_scaling_factor: 8.0


###############################################################################
# Local costmap
###############################################################################
local_costmap:
  global_frame: odom
  robot_base_frame: base_link
  update_frequency: 10.0
  publish_frequency: 5.0
  static_map: false
  rolling_window: true
  width: 10.0
  height: 10.0
  resolution: 0.05
  transform_tolerance: 1.0
  
  # Центрирование локальной карты стоимости относительно робота
  origin_x: -5.0
  origin_y: -5.0
  
  plugins:
    - {name: obstacle_layer, type: "costmap_2d::ObstacleLayer"}
    - {name: inflation_layer, type: "costmap_2d::InflationLayer"}

  obstacle_layer:
    observation_sources: scan
    scan:
      data_type: LaserScan
      topic: /scan
      sensor_frame: lidar_link
      marking: true
      clearing: true
      min_obstacle_height: 0.0
      max_obstacle_height: 2.0
      expected_update_rate: 10.0
      observation_persistence: 0.0
      inf_is_valid: false
      clear_after_reading: true
    obstacle_range: 8.0
    raytrace_range: 10.0
    track_unknown_space: false
    combination_method: 1  # 1 = maximum
    footprint_clearing_enabled: true
    max_obstacle_height: 2.0
  
  inflation_layer:
    inflation_radius: 1.0
    cost_scaling_factor: 5.0
    inflate_unknown: false
    enabled: true
